Sources/app2app/App2AppAuthenticateRequest.swift

```diff
@@ -5,18 +5,20 @@ public struct App2AppAuthenticateRequest {
     public let redirectUri: URL
     public let clientID: String
     public let codeChallenge: String
+    public let state: String?
 
     func toURL() throws -> URL {
         let urlcomponents = URLComponents(string: authorizationEndpoint)
         guard var urlcomponents = urlcomponents else {
             throw AuthgearError.runtimeError("invalid authorizationEndpoint")
         }
-        let query: [String: String] = [
+        var query: [String: String] = [
             "client_id": clientID,
             "redirect_uri": redirectUri.absoluteString,
             "code_challenge_method": Authgear.CodeChallengeMethod,
             "code_challenge": codeChallenge
         ]
+        state.map { query["state"] = $0 }
```

Comment:
The use of `.map` here seems very Kotlin-ish to me. It is more common to use `if let`.

```swift
if let state = state {
  query["state"] = state
}
```